### PR TITLE
Handle double-quotes in the shell similar to real MS-DOS

### LINKED ISSUE #3921: Batch files - echo" command generates an error
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

_No response_

### What Operating System are you using?

Windows 11

### If Other OS, please describe

_No response_

### Relevant hardware info

_No response_

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

echo" command results in "Illegal command" error, while on DOS (tested under Win98), it outputs an empty line

### Steps to reproduce the behaviour.

Explain how to reproduce
Simply type echo" in command prompt and observe the issue.

### Download URL of affected game or software

_No response_

### Your configuration

_No response_

### Provide a Log

_No response_

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

### PR DESCRIPTION
**Pull request by interloper98**
# Description

Real MS-DOS handles the first double quote as a delimiter, for example:

```
C:\> echo"

C:\> echo""
"

C:\> echo "hi"
"hi"

C:\> if"1"=="1" echo matched
matched
```

Where as DOSBox Staging wasn't able to handle this as mentioned in #3921 as well as mentioned by darksky on Discord:

> Does the current version support the `if"` clause and `%1` in batch files? I've noticed quite a few issues with it in the past month but generally assumed it's the game, but now I'm looking into it and I'm noticing potential bugs for example, `if"%1"=="1" goto adlib` results in `Illegal command: if"1"`
> it might not be the cleanest batch file but I believe that worked in DOS

I tested **darksky**'s command and indeed, `if"%1"=="1"` does work in real MS-DOS.

## Related issues

Fixes #3921 as well as the strange space-less format in statements like `if"%1"=="1"`

# Release notes

The command shell now handles double-quotes without preceding spaces similar to real DOS. For example, `ECHO"` will now print an empty line and .BAT files that use squashed syntax, like `if"%1"=="1" goto adlib` will also now work.

# Manual testing

Tested manually against real MS-DOS 6.22 `COMMAND.COM` and also added unit tests.

# Checklist

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [x] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [x] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [x] provided the release notes draft (for significant user-facing changes).
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: src/shell/shell_cmds.cpp [hunk 1]
```diff
@@ -172,27 +172,32 @@ bool DOS_Shell::ExecuteShellCommand(const char* const name, char* arguments)
...
-		if (*line == 32) break;
-		if (*line == '/') break;
-		if (*line == '\t') break;
-		if (*line == '=') break;
-//		if (*line == ':') break; //This breaks drive switching as that is handled at a later stage.
-		if ((*line == '.') ||(*line == '\\')) {  //allow stuff like cd.. and dir.exe cd\kees
-			*cmd_write=0;
+	char* cmd_write = cmd_buffer;
+
+	auto is_cli_delimeter = [](const char c) {
```
Comment (johnnovak) [inline src/shell/shell_cmds.cpp:182]: Typo: delimiter

File: src/shell/shell_cmds.cpp [hunk 2]
```diff
@@ -172,27 +172,32 @@ bool DOS_Shell::ExecuteShellCommand(const char* const name, char* arguments)
...
+		constexpr std::array<char, 7> Delimiters = {'\0', ' ', '/', '\t', '=', '"'};
+		// Note: ':' is also a delimiter, but handling it here breaks
+		//       drive switching as that is handled at a later stage.
+		return contains(Delimiters, c);
+	};
+
+	// Scan forward until we hit the first delimiter
+	while (!is_cli_delimeter(line[0])) {
+		// Allow commands like cd.. and dir.exe cd\kees
```
Comment (johnnovak) [inline src/shell/shell_cmds.cpp:191]: Confusing comment. What is "cd\kees"? Could you clarify and put the example DOS command in double-quotes or list them in a bulleted list?

**Review by johnnovak — APPROVED** [review]
Nice one @interloper98 :tada: Isn't this a regression, though?

File: src/shell/shell_cmds.cpp (hunk 2 above)
Comment (interloper98) [reply src/shell/shell_cmds.cpp:191]: Yup; I just moved that existing comment, but I'll rephrase it to clean it up.

**Comment by interloper98** [comment]
> Nice one @interloper98 🎉 Isn't this a regression, though?

I think it's just a long standing bug. DOSBox SVN doesn't do this right and neither did prior versions of Staging.
